swaps/swaps_service.go

```diff
@@ -0,0 +1,250 @@
+package swaps
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/sha256"
+	"errors"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/BoltzExchange/boltz-client/v2/pkg/boltz"
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/getAlby/hub/config"
+	"github.com/getAlby/hub/events"
+	"github.com/getAlby/hub/lnclient"
+	"github.com/getAlby/hub/logger"
+	"github.com/getAlby/hub/transactions"
+	"github.com/sirupsen/logrus"
+)
+
+type swapsService struct {
+	cancelFn            context.CancelFunc
+	cfg                 config.Config
+	eventPublisher      events.EventPublisher
+	transactionsService transactions.TransactionsService
+}
+
+type SwapsService interface {
+	EnableAutoSwaps(ctx context.Context, lnClient lnclient.LNClient) error
+	StopAutoSwap()
+	ReverseSwap(ctx context.Context, amount uint64, destination string, lnClient lnclient.LNClient) error
+}
+
+func NewSwapsService(cfg config.Config, eventPublisher events.EventPublisher, transactionsService transactions.TransactionsService) *swapsService {
+	return &swapsService{
+		cfg:                 cfg,
+		eventPublisher:      eventPublisher,
+		transactionsService: transactionsService,
+	}
+}
+
+func (svc swapsService) EnableAutoSwaps(ctx context.Context, lnClient lnclient.LNClient) error {
+	// stop any existing swap process
+	svc.StopAutoSwap()
+
+	ctx, cancelFn := context.WithCancel(ctx)
+	swapDestination, _ := svc.cfg.Get(config.AutoSwapDestinationKey, "")
+	balanceThresholdStr, _ := svc.cfg.Get(config.AutoSwapBalanceThresholdKey, "")
+
+	if swapDestination == "" || balanceThresholdStr == "" {
+		cancelFn()
+		return errors.New("auto swap not configured")
+	}
+
+	parsedBalanceThreshold, err := strconv.ParseUint(balanceThresholdStr, 10, 64)
+	if err != nil {
+		cancelFn()
+		return errors.New("invalid auto swap configuration")
+	}
+
+	go func() {
+		ticker := time.NewTicker(1 * time.Hour)
+		for {
+			select {
+			case <-ticker.C:
+				logger.Logger.Info("Checking to see if we can swap")
+				balance, err := lnClient.GetBalances(ctx, false)
+				if err != nil {
+					logger.Logger.WithError(err).Error("Failed to get balance")
+					return
+				}
+				lightningBalance := uint64(balance.Lightning.TotalSpendable)
+				balanceThresholdMilliSats := parsedBalanceThreshold * 1000
+				if lightningBalance >= balanceThresholdMilliSats {
+					// TODO: Change this calcuation
+					amount := lightningBalance - balanceThresholdMilliSats
+					logger.Logger.WithFields(logrus.Fields{
+						"amount":      amount,
+						"destination": swapDestination,
+					}).Info("Initiating swap")
+					// TODO: Should we ourselves add a check that the amount is < 50000
+					err := svc.ReverseSwap(ctx, amount/1000, swapDestination, lnClient)
```

Comment:
Should this be in a goroutine since it blocks calls till this fails/succeeds (I think it's good to block)